bbc/simorgh
# WSTEAM1 1045: Fix 'Maximum 500 words' not announced

Resolves JIRA [WSTEAM1-1045](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1045): 'Maximum 500 words' not announced

Overall changes
======
Use aria-describedby to ensure maximum word limit is announced properly by screen readers.

Code changes
======
- Add logic so that maximum word limit paragraph conditionally renders based on whether a max 'wordLimit' is supplied in the data.
- Add correct aria to textarea if maximum word limit exists
- Add/ update unit tests
- Add story for version of form with a wordLimit

Testing
======
Storybook
1. Go to form example [with maximum word limit](https://wsteam1-1045-max-word-count--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=pages-ugc-page--uploader-form&viewMode=story)
-- check Max word limit paragraph renders
-- check 'aria-describedby' value is added to the textarea with ID that corresponds to the wordLimit.
-- Test on screen readers by naviagting by form fields and ensure max word limit it announced after label.
3. Go to form example [without maximum word limit](https://wsteam1-1045-max-word-count--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=pages-ugc-page--form&viewMode=story). 
-- Check that a Max word limit paragraph does not render.

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
